lib/puppet_x/puppetlabs/powershell/powershell_manager.rb

```diff
@@ -49,6 +71,9 @@ def execute(powershell_code, timeout_ms = nil, working_dir = nil)
 
         out, err = exec_read_result(code, output_ready_event)
 
+        # an error was caught during execution that has invalidated any results
+        return { :exitcode => -1, :stderr => err } if !@usable && out.nil?
```

Comment:
This new failure type should be tested with `unless` and `onlyif` to make sure it works as we intend.